modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/ColocatedSortAggregatePlannerTest.java

```diff
@@ -0,0 +1,382 @@
+                        .and(not(hasAggregate()))
+                        .and(hasGroups())
+                        .and(input(isIndexScan("TEST", "grp0_grp1")))
+                )
+        );
+    }
+
+    @Override
+    protected void checkTestCase12(String sql, IgniteSchema schema) throws Exception {
+        assertPlan(sql, schema,
+                nodeOrAnyChild(isInstanceOf(IgniteColocatedSortAggregate.class)
+                        .and(not(hasAggregate()))
+                        .and(hasGroups())
+                        .and(input(isInstanceOf(IgniteExchange.class)
+                                .and(input(isIndexScan("TEST", "grp0_grp1")))
+                        ))
+                )
+        );
+    }
+
+    @Override
+    protected void checkTestCase13(String sql, IgniteSchema schema) throws Exception {
+        assertPlan(sql, schema,
+                nodeOrAnyChild(isInstanceOf(IgniteColocatedSortAggregate.class)
+                        .and(hasDistinctAggregate())
+                        .and(hasGroups())
+                        .and(input(isInstanceOf(IgniteSort.class)
+                                .and(input(isTableScan("TEST")))
+                        ))
+                )
+        );
+    }
+
+    @Override
+    protected void checkTestCase14(String sql, IgniteSchema schema) throws Exception {
+        assertPlan(sql, schema,
+                nodeOrAnyChild(isInstanceOf(IgniteColocatedSortAggregate.class)
+                        .and(hasDistinctAggregate())
+                        .and(hasGroups())
+                        .and(input(isInstanceOf(IgniteExchange.class)
+                                .and(input(isInstanceOf(IgniteSort.class)
+                                        .and(input(isTableScan("TEST")))
+                                ))
+                        ))
+                )
+        );
+    }
+
+    @Override
+    protected void checkTestCase15(String sql, IgniteSchema schema, String... additionalRules) throws Exception {
+        assertPlan(sql, schema,
+                nodeOrAnyChild(isInstanceOf(IgniteColocatedSortAggregate.class)
+                        .and(input(isIndexScan("TEST", "val0")))
+                ),
+                ArrayUtils.concat(disabledRules(), additionalRules));
+    }
+
+    @Override
+    protected void checkTestCase16(String sql, IgniteSchema schema, String... additionalRules) throws Exception {
+        assertPlan(sql, schema,
+                nodeOrAnyChild(isInstanceOf(IgniteColocatedSortAggregate.class)
+                        .and(input(isInstanceOf(IgniteExchange.class)
+                                .and(input(isIndexScan("TEST", "val0")))
+                        ))
+                ),
+                ArrayUtils.concat(disabledRules(), additionalRules));
+    }
+
+    @Override
+    protected void checkTestCase17(String sql, IgniteSchema schema, RelCollation collation) throws Exception {
```

Comment:
the same as from parent implementation, and seems there are numerous the same issues, is it ok ?